worker/utils/fastlyJob.js

```diff
@@ -21,51 +45,71 @@ class FastlyJobClass {
     let that = this;
     let urlCounter = urlArray.length;
     let purgeMessages = [];
+    const fastly_service_id = environment.getFastlyServiceId();
+  
+      for (let i = 0; i < urlArray.length; i++) {
+        //retrieve surrogate key
+        try {
+          const surrogateKey = await this.retrieveSurrogateKey(urlArray[i])
+          // perform request to purge
+          await this.requestPurgeOfSurrogateKey(surrogateKey)
```

Comment:
awaiting insde a for loop breaks the no-await-in-loop ESLint rule. If we should indeed honor no-await-in-loop, would love your advice on how to refactor it

Reply:
[`Promise.All`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) in general would be the way to batch out async executions for an iterable 👍

Reply:
^^ That isn't necessarily compile + run ready, but would be a good example of how we could do this. 

Batch promise execution in stages like this can be a bit messy - it might make more sense to make a single function that cascades the `retrieveSurrogateKey()` -> `requestPurgeOfSurrogateKey` -> `warmCache` sequence of execution with appropriate awaits, and then batch execution of that function with a `Promise.All()`.